extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheBuildRecorder.java

```diff
@@ -3,21 +3,23 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.Executor;
 
 import org.jboss.logging.Logger;
 
 import io.quarkus.arc.runtime.BeanContainer;
-import io.quarkus.cache.runtime.CacheRepository;
+import io.quarkus.cache.Cache;
+import io.quarkus.cache.runtime.CacheManagerImpl;
 import io.quarkus.runtime.annotations.Recorder;
 
 @Recorder
 public class CaffeineCacheBuildRecorder {
 
     private static final Logger LOGGER = Logger.getLogger(CaffeineCacheBuildRecorder.class);
 
-    public void buildCaches(BeanContainer beanContainer, Set<CaffeineCacheInfo> cacheInfos) {
+    public void buildCaches(Set<CaffeineCacheInfo> cacheInfos, Executor executor, BeanContainer beanContainer) {
```

Comment:
Where does thaty executor come from in practice?

Reply:
The executor-related logic changed a bit since your review because of #8227 merge on `master`.

Here's the executor origin with the latest version:
- If a `ManagedExecutor` instance is available in the application (which requires `quarkus-smallrye-context-propagation` to be present), then it is used here.
- Otherwise, the executor provided by `ExecutorBuildItem` is used (instead of `ForkJoinPool#commonPool` before this PR).

I am not sure using the executor from `ExecutorBuildItem` is the best choice though. This needs to be discussed here.

Reply:
There's no need for an executor anymore so I'm resolving this.